src/tallies/filter_legendre.cpp

```diff
@@ -10,7 +10,13 @@ namespace openmc {
 void
 LegendreFilter::from_xml(pugi::xml_node node)
 {
-  order_ = std::stoi(get_node_value(node, "order"));
+  this->set_order(std::stoi(get_node_value(node, "order")));
+}
+
+void
+LegendreFilter::set_order(int order)
+{
+  order_ = order;
```

Comment:
Maybe add a check here (and in the other functional expansion filters) to make sure the order is >= 0.